web/src/routes/results/[eid]/+page.svelte

```diff
@@ -0,0 +1,259 @@
+<script lang="ts">
+	import Button from "$lib/button.svelte";
+	import Dialog from "$lib/dialog.svelte";
+	import Panel from "$lib/panel.svelte";
+
+	import {
+		type ElectionOutcome,
+		type ElectionOutcomeResult,
+		elections,
+		error,
+		fetching,
+		user,
+	} from "../../../store";
+	import { goto } from "$app/navigation";
+	import { API } from "$lib/endpoints";
+	import { _getCurrentElection, _getElections } from "../../+layout";
+
+	export let data: { data: ElectionOutcome };
+
+	$: electionOutcome = data.data;
+
+	$: election = electionOutcome.election;
+	$: resultsByRound = electionOutcome.results.reduce((res, cur) => {
+		return {
+			...res,
+			[cur.round]: res[cur.round]
+				? [...res[cur.round], cur].sort((a, b) =>
+						a.voteCount != b.voteCount ? b.voteCount - a.voteCount : b.isRejected ? -1 : 1,
+				  )
+				: [cur],
+		};
+	}, {} as { [key: string]: Array<ElectionOutcomeResult> });
+	$: winner = electionOutcome.results
+		.filter((e) => e.isElected)
+		.map((e) => e.name)
+		.join(", ");
+	$: date = new Date(electionOutcome.date).toUTCString();
+
+	let deleteElectionDialog: HTMLDialogElement;
+	const deleteElection = async (id: number) => {
+		$fetching = true;
+		const response = await fetch(API.ADMIN_ELECTION, {
+			method: "DELETE",
+			body: JSON.stringify({ id }),
+		});
+
+		if (response.ok) {
+			elections.set(await _getElections());
+			goto("/");
+		} else {
+			$error = new Error(await response.text());
+		}
+		$fetching = false;
+	};
+
+	const publishElectionResults = async (id: number, published: boolean) => {
+		$fetching = true;
+		const response = await fetch(API.ADMIN_ELECTION_PUBLISH, {
+			method: "POST",
+			body: JSON.stringify({ id, published }),
+		});
+
+		if (response.ok) {
+			electionOutcome.isPublished = published;
+		} else {
+			$error = new Error(await response.text());
+		}
+		$fetching = false;
+	};
+</script>
+
+<svelte:head>
+	<title>Outcome of election for {election.roleName}</title>
+</svelte:head>
+
+<Panel title={`Outcome of election for ${election.roleName}`}>
+	<p>{election.description}</p>
+</Panel>
+
+{#if $user.isAdmin && !electionOutcome.isPublished}
+	<Panel kind="emphasis">
+		<div class="unpublished-callout">
+			<p>
+				The results for <b>{election.roleName}</b> have not been published. While they remain unpublished,
+				only administrators may view the results.
+			</p>
+			<div class="button">

```

Comment:
Not sure if this wrapper div is needed? See comment below about using `grid` instead